src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.VectorSearch.cs

```diff
@@ -293,26 +293,20 @@ internal static (VectorValue? SingleVector, VectorValue[] MultiVector) GetEmbedd
             
             var embeddingsTaskId = new EmbeddingsGenerationTaskIdentifier(embeddingsGenerationTaskIdentifier);
 
-            if (database.AiIntegrations.TryGetConnectionStringIdByEmbeddingsGenerationTask(embeddingsTaskId, out var connectionStringId) == false)
-                throw new InvalidQueryException($"Couldn't find Embeddings Generation task with '{embeddingsGenerationTaskIdentifier}' identifier");
 
-            var sourceEmbeddingType = VectorEmbeddingType.Single;
-
-            if (builderParameters.Index.DocumentDatabase.AiIntegrations.TryGetEmbeddingsGenerationConfiguration(embeddingsTaskId,
-                    out var embeddingsGenerationConfiguration))
-            {
-                sourceEmbeddingType = embeddingsGenerationConfiguration.Quantization;
-            }
+            var embeddingsGenerator = database.EmbeddingsGeneratorQueries;
+            
+            var sourceEmbeddingType = embeddingsGenerator.GetQuantizationOf(embeddingsTaskId);
 
             var destinationEmbeddingType = vectorOptions?.DestinationEmbeddingType ?? sourceEmbeddingType;
             
-            IEmbeddingValue[] embeddingValues;
+            ReadOnlyMemory<ReadOnlyMemory<byte>> embeddingValues;
 
             switch (valueType)
             {
                 case ValueTokenType.String:
-                    embeddingValues = database.AiIntegrations.Embeddings
-                        .GetEmbeddingsForQueryAsync(builderParameters.DocumentsContext, connectionStringId, embeddingsTaskId, value.ToString())
+                    embeddingValues = embeddingsGenerator
```

Comment:
Warning about `ValueTask.Result` usage - it should never be performed (https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1?view=net-9.0)

Reply:
Fixed